[Fix] Possible fix for matter-test-scripts issue #228: Remove PICS from RVC tests

j-ororke opened the pull request:
Following changes were applied and used:
- Removed PICS from python RVC Clean and RVC Run tests
-- Replaced with commands and attributes gathered from DUT
- Added checking command line arguments for "--app-pid" to enable self.is_ci variable in RVC test scripts in place of PICS "PICS_SDK_CI_ONLY" check.

#### Testing

j-ororke committed:
Possible fix for mts issue #228:
- Removed PICS from python RVC Clean and RVC Run tests
- Replaced with commands and attributes gathered from DUT.

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Removed unused variable as no longer needed:
- Removed "generated_command_list" variable as not needed in TC_RVCRUNM_2_2 test module

j-ororke committed:
Removed unused variable:
- Removed unused variable "RVCRun_gencmd_list"

cecille commented on src/python_testing/TC_RVCCLEANM_2_1.py:
```diff
@@ -84,19 +84,40 @@ async def test_TC_RVCCLEANM_2_1(self):
-            app_pid = self.matter_test_config.app_pid
-            if app_pid == 0:
-                asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set")
+        app_pid = self.matter_test_config.app_pid
```
You can just leave this CI check in for now. The question of how to determine whether an app is running in the CI is fairly open, but this PICS code is kind of meta. The goal for these issues is to remove the PICS codes for things that are represented on the device already (clusters, features, attributes, commands etc). This one is fine.

cecille commented on src/python_testing/TC_RVCCLEANM_2_1.py:
```diff
@@ -84,19 +84,40 @@ async def test_TC_RVCCLEANM_2_1(self):
-            if app_pid == 0:
-                asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set")
+        app_pid = self.matter_test_config.app_pid
+        if app_pid != 0:
```
I know this isn't your code, but I'm also not sure this is correct. If the flag isn't set, won't app_pid be None?

j-ororke replied:
Hi Cecille,
So when I run this in the Raspberry pi Dev container it was returning a 0 for app_pid if --app-pid is not provided in the CLI arguments. 
In the matter_testing_support module it appears to set the possible variables to 0 if they are to be gathered by argparse and not found in the provided command line arguments:
- config.app_pid = 0 if args.app_pid is None else args.app_pid

j-ororke committed:
Reverted changes to self.is_ci check:
- Reverted the changes done to the self.is_ci if check block that was made as it was not passing through CI/CD tests.

cecille commented on src/python_testing/TC_RVCCLEANM_1_2.py:
```diff
@@ -57,11 +57,16 @@ def pics_TC_RVCCLEANM_1_2(self) -> list[str]:
         self.print_step(1, "Commissioning, already done")
 
-        if self.check_pics("RVCCLEANM.S.A0000"):
+        if supported_modes_attr_id in attribute_list:
```
Q for the RVC folks - this is mandatory, right? Can we just remove this check?

hicklin replied:
Yes it is. Same of `A0001`. Should this check be made somewhere else, or is it dealt by the data model?

cecille commented on src/python_testing/TC_RVCCLEANM_1_2.py:
```diff
@@ -124,7 +129,7 @@ async def test_TC_RVCCLEANM_1_2(self):
                                 "At least one ModeOptionsStruct entry must include either the Vacuum or Mop mode tag")
 
-        if self.check_pics("RVCCLEANM.S.A0001"):
+        if current_mode_attr_id in attribute_list:
```
ditto here

cecille commented on src/python_testing/TC_RVCCLEANM_2_1.py:
```diff
@@ -87,16 +87,39 @@ async def test_TC_RVCCLEANM_2_1(self):
+        chg_mode_cmd_id = commands.ChangeToMode.command_id
+        chg_rsp_cmd_id = commands.ChangeToModeResponse.command_id
+
+        if supported_modes_attr_id not in attribute_list:
```
same q here - these are all mandatory, would it make sense to just remove these given that we have conformance checks now?

hicklin replied:
If there are other ways of checking this, yes.

cecille reviewed (commented):
General Q for the RVC folks - do you think we need these checks at all, or should we just remove the checks entirely?

In 1.4 we're moving to a system where you need to pass conformance and PICS checks before running any other tests, so in theory you won't see non-conformant devices come through to here. Or would you prefer to leave these in as an additional check?

j-ororke committed:
Merge branch 'master' into remove_PICS_from_RVC_tests

hicklin commented on src/python_testing/TC_RVCCLEANM_2_1.py:
```diff
@@ -92,16 +92,39 @@ async def test_TC_RVCCLEANM_2_1(self):
-        asserts.assert_true(self.check_pics("RVCCLEANM.S.C01.Tx"), "RVCCLEANM.S.C01.Tx must be supported")
-
-        attributes = Clusters.RvcCleanMode.Attributes
+            if app_pid != 0:
```
I like this. Should this be indented one less together with like 94 and line 93 be removed?

Same comment in clean 2.2 and run 2.1 and 2.2.

j-ororke replied:
I believe you are correct, I have unindented lines 95 and 94 as the self.is_ci condition check on line 93 does not appear to be needed after running some tests without it. I have done the same in clean 2.2 and run 2.1 and 2.2 test modules as well.

j-ororke replied:
Sorry, I have re-added the if self.is_ci checks as they appear to be needed for these tests

hicklin commented on src/python_testing/TC_RVCCLEANM_2_2.py:
```diff
@@ -94,13 +94,23 @@ async def test_TC_RVCCLEANM_2_2(self):
+            asserts.fail("Supported modes needs to be supported attribute")
+
+        if current_mode_attr_id not in attribute_list:
+            asserts.fail("Current mode needs to be supported attribute")
```
I guess these can be removed as well now. Same in the following scripts.

j-ororke replied:
Thank you, I have removed these conditional checks for current mode and supported mode attributes from the RVC Clean and Run test modules.

j-ororke committed:
Updated RVC Clean and Run test modules:
- Removed initial step for if self.is_ci check as does not appear to be needed
- Removed attribute checks as supported and current mode attributes are mandatory

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Linting fixes in RVCCLEAN 2_2 and RVCRUN 1_2 modules

cecille reviewed (commented):
I think the CI checks still need to be put back, but this looks otherwise correct to me. 

@hicklin - would you mind taking another look?

j-ororke committed:
Merge branch 'master' into remove_PICS_from_RVC_tests

j-ororke committed:
Merge branch 'master' into remove_PICS_from_RVC_tests

j-ororke committed:
Updated RVC Clean and RVC Run test modules:
- Updated to restore using if self.is_cli checks

restyled-commits committed:
Restyled by autopep8

hicklin commented on src/python_testing/TC_RVCCLEANM_2_1.py:
```diff
@@ -96,14 +96,24 @@ async def test_TC_RVCCLEANM_2_1(self):
             app_pid = self.matter_test_config.app_pid
             if app_pid == 0:
                 asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set")
-            self.app_pipe = self.app_pipe + str(app_pid)
```
I think that this is needed as `write_to_app_pipe` is used later without specifying the `app_pipe_name`. See [here](https://github.com/project-chip/connectedhomeip/blob/f212d601738728d58e88ada3bc7e65b84cdfe7f0/src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py#L1028).

j-ororke replied:
Thank you @hicklin,
You are correct that was definitely needed in there and not sure how I had deleted it. 
However, I have returned it to the code now.

hicklin commented on src/python_testing/TC_RVCCLEANM_2_2.py:
```diff
@@ -107,15 +107,11 @@ async def test_TC_RVCCLEANM_2_2(self):
             app_pid = self.matter_test_config.app_pid
             if app_pid == 0:
-                asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set.c")
-            self.app_pipe = self.app_pipe + str(app_pid)
```
I think that this is needed as `write_to_app_pipe` is used later without specifying the `app_pipe_name`.

j-ororke replied:
Thank you, you are indeed correct that was definitely needed in there and not sure how I had deleted it.
However, I have returned it to the code now.

hicklin commented on src/python_testing/TC_RVCRUNM_2_1.py:
```diff
@@ -90,15 +90,25 @@ async def test_TC_RVCRUNM_2_1(self):
             app_pid = self.matter_test_config.app_pid
             if app_pid == 0:
-                asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set.c")
-            self.app_pipe = self.app_pipe + str(app_pid)
```
this may be needed.

j-ororke replied:
Thank you, you are correct that was definitely needed in there. I have returned it to its rightful place here now.

hicklin commented on src/python_testing/TC_RVCRUNM_2_2.py:
```diff
@@ -135,21 +135,26 @@ async def test_TC_RVCRUNM_2_2(self):
             app_pid = self.matter_test_config.app_pid
             if app_pid == 0:
-                asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set.c")
-            self.app_pipe = self.app_pipe + str(app_pid)
```
This may be needed

j-ororke replied:
Thank you, you are correct that was definitely needed in there. I have returned it to its rightful place here now.

hicklin reviewed (changes requested):
Looks good. I think there is a small mistake in removing the setting of  `self.app_pipe`.

j-ororke committed:
Updating RVCRUNM_2 and RVCCLEANM_2 test modules:
- Adding back in the self.app_pipe code as it appears to have been accidentally removed

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into remove_PICS_from_RVC_tests

j-ororke committed:
Merge branch 'master' into remove_PICS_from_RVC_tests

j-ororke committed:
Updating TC_RVCCLEANM_2_1, RVCRUNM_2_1, and RVCRUNM_2_2 test modules:
- Updating to using command_guard() and attribute_guard() where applicable

j-ororke committed:
Updating TC_RVCCLEANM_2_1 and TC_RVCRUNM_2_2 test modules:
- Resolving linting errors

j-ororke committed:
Update TC_RVCRUNM_2_2 python test module:
- Resolving linting error

j-ororke committed:
Updating TC_RVCRUNM_2_2 python test module:
- Resolving linting errors

j-ororke committed:
Merge branch 'master' into remove_PICS_from_RVC_tests
